[test_vision_attention.py] | (Tier 2) Models Unit [scheduled] | models-unit-tests / Qwen2.5-VL-72B unit tests [wh_llmbox_perf]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] Qwen2.5-VL-72B: writer_windowed.cpp kernel compilation failure — no matching function for write_block()
<!-- AUTO-TRIAGE-SUMMARY-END -->

**Workflow:** (Tier 2) Models Unit [scheduled]
**Job:** models-unit-tests / Qwen2.5-VL-72B unit tests [wh_llmbox_perf]

### Regression timeline

- **Last passing:** commit `85975a50d8f4` — 2026-06-10T10:28:52Z — https://github.com/tenstorrent/tt-metal/actions/runs/27270021788
- **First failing:** commit `60ea807b6ba8` — 2026-06-11T01:49:52Z — https://github.com/tenstorrent/tt-metal/actions/runs/27318415519
- **Most recent failing:** commit `46e6f47968be` — 2026-06-14T13:32:55Z — https://github.com/tenstorrent/tt-metal/actions/runs/27500470408

### Failing job URLs (last 2 runs)

- https://github.com/tenstorrent/tt-metal/actions/runs/27500470408/job/81282806018
- https://github.com/tenstorrent/tt-metal/actions/runs/27485098256/job/81240238572

### Failing test path(s)

- `models/demos/qwen25_vl/tests/test_vision_attention.py::test_vision_attention_inference[wormhole_b0-device_params0-mesh_device0]`
- `models/demos/qwen25_vl/tests/test_vision_block.py::test_vision_block_inference[wormhole_b0-device_params0-mesh_device0]` (secondary, hit same error)

Test command:
```
pytest --timeout 600 models/demos/qwen25_vl/tests/ \
  --ignore=models/demos/qwen25_vl/tests/test_ci_dispatch.py \
  --ignore=models/demos/qwen25_vl/tests/conftest.py
```

### Error excerpt (terminal failure from logs)

```
##[error]test_vision_attention_inference[wormhole_b0-device_params0-mesh_device0]
RuntimeError: TT_THROW @ /project/tt_metal/impl/program/program.cpp:255: tt::exception
Failed to generate binaries for writer_windowed TT_THROW @ /project/tt_metal/jit_build/build.cpp:67: tt::exception
brisc build failed. Log: In file included from ../kernel_includes.hpp:1,
/work/ttnn/cpp/ttnn/operations/transformer/sdpa_windowed/device/kernels/dataflow/writer_windowed.cpp:77:28: error: no matching function for call to 'write_block(const TensorAccessor<tensor_accessor::DistributionSpec<0, 0, tensor_accessor::ArrayStaticWrapper<long unsigned int>, tensor_accessor::ArrayStaticWrapper<long unsigned int>, tensor_accessor::ArrayStaticWrapper<short unsigned int>, true, true> >&, const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&, uint32_t&, const uint32_t&, const uint32_t&)'
Fatal Python error: Segmentation fault
##[error]Process completed with exit code 139.
```

### Reproduction steps

1. Check out the failing commit (`46e6f47968be` or later).
2. Set up the wormhole_b0 environment (4-device mesh, `wh_llmbox_perf` runner).
3. Install the ttnn wheel and run:
   ```bash
   pytest --timeout 600 models/demos/qwen25_vl/tests/ \
     --ignore=models/demos/qwen25_vl/tests/test_ci_dispatch.py \
     --ignore=models/demos/qwen25_vl/tests/conftest.py
   ```
4. Observe the kernel JIT compile failure for `writer_windowed` followed by a segfault.

### Notes

The root cause is a **C++ overload resolution failure** in the windowed SDPA kernel:
- `writer_windowed.cpp:77` calls `write_block()` with 8 arguments.
- The available `write_block()` candidates in `dataflow_common.hpp` do not match this signature (candidates require a different number/type of arguments).
- This causes `TT_THROW: brisc build failed` → `TT_THROW: Failed to generate binaries for writer_windowed` → `RuntimeError` in the test → segfault during pytest teardown (exit code 139).

The regression was introduced between `85975a50d8f4` (2026-06-10) and `60ea807b6ba8` (2026-06-11), suggesting a change to either `write_block()` signature in `dataflow_common.hpp` or to `writer_windowed.cpp` call site broke compatibility.

**Determinism check:**
- **Run 1** (`27500470408`): `##[error]test_vision_attention_inference[wormhole_b0-device_params0-mesh_device0]` + `writer_windowed.cpp:77:28: error: no matching function for call to 'write_block(...)'` + `##[error]Process completed with exit code 139.`
- **Run 2** (`27485098256`): `##[error]test_vision_attention_inference[wormhole_b0-device_params0-mesh_device0]` + `writer_windowed.cpp:77:28: error: no matching function for call to 'write_block(...)'` + `##[error]Process completed with exit code 139.`
- Both runs show identical kernel, file path, line number, and function signature in the error.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: (Tier 2) Models Unit [scheduled]`
`Auto-triage-job-name: models-unit-tests / Qwen2.5-VL-72B unit tests [wh_llmbox_perf]`
<!-- AUTO-TRIAGE-METADATA-END -->

--- a/ttnn/cpp/ttnn/operations/transformer/sdpa_windowed/device/kernels/dataflow/writer_windowed.cpp
+++ b/ttnn/cpp/ttnn/operations/transformer/sdpa_windowed/device/kernels/dataflow/writer_windowed.cpp
@@ -49,6 +49,8 @@ void kernel_main() {
     constexpr uint32_t barrier_threshold = get_barrier_read_threshold<tile_bytes, num_cores>();
     uint32_t barrier_count = 0;
 
+    Noc noc;
+
     constexpr uint32_t cb_identity_scale_in = tt::CBIndex::c_5;
     constexpr uint32_t cb_col_identity = tt::CBIndex::c_7;
 
@@ -75,6 +77,7 @@ void kernel_main() {
                 const uint32_t out_row_tile_count = out_row_end_tile - out_row_start_tile;
                 uint32_t out_tile_id = out_tile_shape.id_of(nb, nq, out_row_start_tile, 0);
                 write_block(
+                    noc,
                     out_writer,
                     cb_out,
                     out_chunk_tiles,

